### DISCUSSION TITLE
How we can add support for ThorVG?

### DISCUSSION THREAD

**Discussion by paulocoutinhox** [axmolengine/axmol#2042] [General]
Hi,

How we can add support for ThorVG?

Ref:
https://github.com/thorvg/thorvg

Thanks.

**Comment by NgVThangBz** [comment]
I'm also interested in it, currently I'm making an application that needs to read svg files and I display it manually using DrawNode. 
It would be nice if axmol had support for SVG files

**Comment by aismann** [comment]
Here you can see a preview of my current implementation of SVG
![image](https://github.com/user-attachments/assets/f4ea14ae-5139-4e67-b406-796ecb41f96d)
![image](https://github.com/user-attachments/assets/9c65d84e-8f70-47bf-9979-4f91053bd31e)
![image](https://github.com/user-attachments/assets/74f70ed8-18ec-4c05-a66f-d68fca68d0d0)

**Reply by NgVThangBz** [reply]
you use thorVG or just use DrawNode

**Reply by aismann** [reply]
DrawNode

**Reply by NgVThangBz** [reply]
ah, yes 
my application also uses the same way, but there are still some performance issues

![photo_2024-09-06_15-28-06](https://github.com/user-attachments/assets/ebc63bef-b3c8-42d9-961a-dad3f149eea8)

**Reply by aismann** [reply]
can i get your SVGs?
for performance testing?

**Reply by aismann** [reply]
Here are the first "filled" examples (but thats buggy at the moment)
Performance is good
![image](https://github.com/user-attachments/assets/b9ea8cfd-058e-4f2f-bfa1-10c4d07fc513)
![image](https://github.com/user-attachments/assets/27a729f4-1a70-41c6-90d8-8cec29e5f74e)

**Reply by NgVThangBz** [reply]
here is my svg, you can check it: 
[Sofa_01_B.svg.zip](https://github.com/user-attachments/files/16905591/Sofa_01_B.svg.zip)

**Reply by aismann** [reply]
Looks it correct? (I scaled it)
How is the performance?
![image](https://github.com/user-attachments/assets/adce8380-065a-4db3-b888-b6a1235a646b)

**Reply by NgVThangBz** [reply]
now you just draw line
When you need to display colors accurately, you will have to deal with polygon, triangulation (DrawNode has it)
and this is a static object, if it moves, need to clear all and reDraw

**Reply by aismann** [reply]
> now you just draw line When you need to display colors accurately, you will have to deal with polygon, triangulation (DrawNode has it) and this is a static object, if it moves, need to clear all and reDraw

The fill routine is not working well...but look here: (random color)
![image](https://github.com/user-attachments/assets/2707086b-a2bd-420c-9010-d3d737a23fc1)
and here its rotated:
![image](https://github.com/user-attachments/assets/28514b27-fc10-4b4c-b0f5-6975b3b145bb)

**Reply by aismann** [reply]
How looks your sofa?

**Reply by NgVThangBz** [reply]
It's not correct scale, but it should look like this
![image](https://github.com/user-attachments/assets/36b9bb88-5150-41af-a0d5-f87cc09b73f7)

**Comment by aismann** [comment]
Performance looks good.
Filling is working well!
Color is working.
Rotation has some "mistakes".
Scale is working.
Gradient color not implemented.
![image](https://github.com/user-attachments/assets/830576a6-82ba-4e38-8ac9-220c85014961)
![image](https://github.com/user-attachments/assets/d194b420-1ad1-480a-bf8b-441754587bf8)
![image](https://github.com/user-attachments/assets/c100de64-5ff2-4594-a7ac-681431b57069)

**Reply by paulocoutinhox** [reply]
Wow, very good. You think that it can replace our PNG files? Anything against it?

**Reply by aismann** [reply]
I belive this will be a closed source library (at the moment).
SVG is not PNG => but it can internaly convert to any other "pixel-format images" like PNG.
The goal is to use it as an `ax::Sprite` on axmol.

**Reply by paulocoutinhox** [reply]
Ah. I think that you are using ThorVG.

**Reply by aismann** [reply]
> Ah. I think that you are using ThorVG.

No. Its native DrawNode
